### DISCUSSION TITLE
Add a profile for lanraragi (nodejs)

### DISCUSSION THREAD

**Discussion by Crehet** [netblue30/firejail#5501] [Q&A]
[This repo](https://github.com/Difegue/LANraragi)
I'm only making use of the npm.profile for said project. Would said profile be enough to 'properly' sandbox it?

**Comment by kmk3** [comment]
@Crehet [on Dec 3](https://github.com/netblue30/firejail/discussions/5501#discussion-4623748):

> [This repo](https://github.com/Difegue/LANraragi)
> 
> I'm only making use of the npm.profile for said project. Would said profile
> be enough to 'properly' sandbox it?

How is the program executed? `node foo` or `foo` directly?

It looks like npm.profile/nodejs-common.profile are mainly intended for
installing npm packages for development, not for running nodejs-based programs
with GUIs.  Also, it is not a whitelisting profile and so it might leave many
files in the home directory still accessible.  See the commented `whitelist`
commands in the included nodejs-common.profile.

To see what files a profile allows/blocks, run a shell with the profile and try
seeing what files are present and what are their permissions.  Examples:

Running commands directly:

```
firejail --profile=npm ls -l ~/
firejail --profile=npm find ~/ | LC_ALL=C sort
```

Sandboxed shell:

```
firejail --profile=npm /bin/bash
```

So it would probably make more sense to create a profile for this program.  The
nodejs-common.profile could be used as a base and then the needed configuration
paths could be whitelisted, for example.  See also `--build` in the firejail(1)
man page.

**Reply by Crehet** [reply]
>How is the program executed? node foo or foo directly?

Starting `redis-server` comes first as it is required.

```console
$ redis-server
$ cd /home/$USER/lanraragi
$ npm start
$ firefox # then I type `localhost:3000` in the urlbar.
```

The above commands are the _normal_ way to start lanraragi without any sandboxing. Of course I have a nodejs-common.profile in my config folder with all the whitelist suggestions uncommented (because why not? im testing it afterall)


My installation steps were like this:

```console
$ git clone -b master http://github.com/Difegue/LANraragi /home/$USER/lanraragi
$ cd /home/$USER/lanraragi && sudo npm run lanraragi-installer install-full
```

`sudo` is not needed because of this package [perl-config-autoconf](https://archlinux.org/packages/extra/any/perl-config-autoconf) and then I use `env` variable
`export PERL5LIB=~/perl5/lib/perl5` before running the installer
If dependencies are relevant I apologize for not mentioning them.

---

Edit by @kmk3: Formatting.
